v2/internal/frontend/desktop/linux/window.go

```diff
@@ -736,7 +708,9 @@ func (w *Window) SetDecorated(frameless bool) {
 }
 
 func (w *Window) SetTitle(title string) {
-	C.SetTitle(w.asGTKWindow(), C.CString(title))
+	newTitle := C.CString(title)
+	defer C.free(unsafe.Pointer(newTitle))
+	C.gtk_window_set_title(w.asGTKWindow(), newTitle)
```

Comment:
I think this operation needs to be on the main thread. If not, it can bork on ARM64 arch. The original code created the C.CString which was freed at the end of the `SetTitle` C method, which is what this change does. What was the original error you were seeing?

Reply:
From tracing through and testing `runtime.WindowSetTitle`from both JS and Go, I couldn't find a reason to hop across to the main thread. This code worked consistently both at startup and runtime.

However, this is not my area of expertise, so if forcing main thread use is a must, especially for ARM64, I'm game to try and reintroduce it.

The errors I'm seeing at startup are...

```
(wails-v2-test:62686): Gtk-CRITICAL **: 22:55:36.314: gtk_window_set_title: assertion 'GTK_IS_WINDOW (window)' failed
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xfffffffffffffffb pc=0x7f9f5e73ea95]
```

or

```
fatal error: unexpected signal during runtime execution
[signal SIGBUS: bus error code=0x80 addr=0x0 pc=0x7f2a0dbdd77a]
```

The failed assertion by `gtk_window_set_title` made me think there was a race condition due to `SetTitle` being called before the `gtk_main()` call. Commenting out the `result.SetTitle(appoptions.Title)`call in `NewWindow` stopped the errors on startup. Like with the `StartDrag` fix below, I tried various ways of guarding against the window reference not be what it should be, but found stripping right back seemed to do the job.